testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/client/RequestNamedQueryParameterTest.java

```diff
@@ -0,0 +1,186 @@
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.resteasy.annotations.jaxrs.QueryParam;
+import org.jboss.resteasy.client.jaxrs.ClientHttpEngine;
+import org.jboss.resteasy.client.jaxrs.ResteasyClient;
+import org.jboss.resteasy.client.jaxrs.ResteasyClientBuilder;
+import org.jboss.resteasy.client.jaxrs.internal.AbortedResponse;
+import org.jboss.resteasy.client.jaxrs.internal.ClientInvocation;
+import org.jboss.resteasy.core.ServerResponse;
+import org.jboss.resteasy.utils.TestUtil;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author <a href="mailto:dkafetzi@redhat.com">Dimitris Kafetzis</a>
+ * @tpSubChapter Resteasy-client
+ * @tpChapter Integration tests
+ * @tpSince RESTEasy 6.3.0
+ */
+@RunWith(Arquillian.class)
+@RunAsClient
+public class RequestNamedQueryParameterTest extends ClientTestBase {
+
+    @Path("someResource")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Consumes(MediaType.APPLICATION_JSON)
+    public interface SomeResource {
+
+        @GET
+        void methodWithLists(@QueryParam("listA") List<String> listA, @QueryParam("listB") List<String> listB,
+                @QueryParam("listC") List<String> listC);
+    }
+
+    @Deployment
+    public static Archive<?> deploy() {
+        WebArchive war = TestUtil.prepareArchive(ClientInvocationBuilderTest.class.getSimpleName());
+        war.addClass(SomeResource.class);
+        war.addClass(ClientTestBase.class);
+        return war;
+    }
+
+    /**
+     * @tpTestDetails Check if empty named query parameters of a request are properly handled.
+     * @tpPassCrit The query for this request invocation should be empty.
+     *             The empty Named Query Parameters should be Ignored
+     *             (no stray or duplicate '&'s should be present)
+     * @tpSince RESTEasy 6.3.0
+     */
+    @Test
+    public void testWithEmptyNamedQueryParameters() {
+        ResteasyClientBuilder builder = (ResteasyClientBuilder) ClientBuilder.newBuilder();
+        ResteasyClient client = builder.httpEngine(new ClientHttpEngine() {
+            @Override
+            public Response invoke(Invocation request) {
+                Assert.assertEquals("", ((ClientInvocation) request).getUri().getQuery());
+                return new AbortedResponse(null, new ServerResponse());
+            }
+
+            @Override
+            public SSLContext getSslContext() {
+                return null;
+            }
+
+            @Override
+            public HostnameVerifier getHostnameVerifier() {
+                return null;
+            }
+
+            @Override
+            public void close() {
+            }
+        }).build();
+
+        try {
+            client.target("http://a.place.org").proxy(SomeResource.class).methodWithLists(List.of(), List.of(), List.of());
```

Comment:
We should use localhost in all the places here. While this shouldn't be resolved given the above, this is a potential exploit point.